x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/DataNodeRequestSenderIT.java

```diff
@@ -52,17 +50,19 @@ public void testSearchWhileRelocating() throws InterruptedException {
 
         // start background searches
         var stopped = new AtomicBoolean(false);
-        var queries = new LongAdder();
         var threads = new Thread[randomIntBetween(1, 5)];
         for (int i = 0; i < threads.length; i++) {
             threads[i] = new Thread(() -> {
                 while (stopped.get() == false) {
                     try (EsqlQueryResponse resp = run("FROM index-1")) {
                         assertThat(getValuesList(resp), hasSize(docs));
+                    } catch (Exception | AssertionError e) {
+                        logger.warn("Query failed with exception", e);
+                        stopped.set(true);
+                        throw e;
                     }
-                    queries.increment();
                 }
-            });
+            }, "testSearchWhileRelocating");
```

Comment:
Would it help to have the `i` in the name?

Reply:
I would say it is not to important as long as we can identify this is the the source of the threads if they ever picked up by leak detector

Reply:
Although, there is no disadvantage in having it, will add shortly.